### DISCUSSION TITLE
with a SQL-like querying language for Khepri

### DISCUSSION THREAD

**Discussion by leeyisoft** [rabbitmq/khepri#202] [Ideas]
I don't have the time and energy to realize this at the moment, so let me record it first

Cursory Google search, found a few related library, do not know whether useful

https://github.com/basho/riak_ql   SQL query language for Riak

https://github.com/ddosia/mekao

https://github.com/artemeff/eql

https://github.com/devinus/sqerl

**Comment by michaelklishin** [comment]
I don't think Khepri is at a point where its tiny team can begin looking into nice-to-haves like this.

The libraries listed produce SQL from Erlang terms or an Erlang-based DSL. Implementing support for a SQL-like language needs a parser for it, and a query executor/planner. That's something completely different from *generating* SQL.

**Comment by lukebakken** [comment]
@leeyisoft you should also look into [`riak_ql`](https://github.com/basho/riak_ql)

**Reply by leeyisoft** [reply]
[riak_ql](https://github.com/basho/riak_ql) is really a great library that meets this requirement. I will study it carefully when I have time in the future. Thank you very much for your reply

**Comment by leeyisoft** [comment]
I agree with you(@michaelklishin ): "Implementing support for a SQL-like language needs a parser for it, and a query executor/planner.",

I have no knowledge of SQL executor/parser. I am a white who is interested in these technologies but has no time and energy to learn them deeply.

Maybe this feature is optional now, but as khepri grows, it's probably needed. I'll put it on the discussion board
